apps/dot_ext/views/authorization.py

```diff
@@ -141,6 +142,14 @@ def form_valid(self, form):
             # "code_challenge_method": form.cleaned_data.get("code_challenge_method", None),
         }
 
+        if switch_is_active('require_pkce'):
+            if not form.cleaned_data.get("code_challenge") or not form.cleaned_data.get("code_challenge_method"):
+                response = {"error": "Missing Required Parameter(s): code_challenge, code_challenge_method"}
+                return JsonResponse(response, status=400)
+        if switch_is_active('require_state') and not form.cleaned_data.get("state"):
+            response = {"error": "Missing Required Parameter(s): state"}
+            return JsonResponse(response, status=400)
```

Comment:
Let's make this closer to the response when you provide invalid scopes in the request:
`return JsonResponse({"status_code": 400, "message": "Invalid scopes."}, status=400)`